### PR TITLE
COMMERCE-11535

### PR DESCRIPTION
**Pull request by brianikim**
https://issues.liferay.com/browse/COMMERCE-11535
https://issues.liferay.com/browse/COMMERCE-11523
https://issues.liferay.com/browse/COMMERCE-11556

@andrea-ale-sbarra , tied to LPP

Only commits: [5358b27](https://github.com/liferay-commerce/liferay-portal/pull/3796/commits/5358b270271278ff873cfdae0047218945cb3e71) and [ae1c864](https://github.com/liferay-commerce/liferay-portal/pull/3796/commits/ae1c8643605344845d3d29746a078a62d476cfee) need to be reviewed.

Sincerely,
Brian I. Kim

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
@brianikim Please fix the conflicts, thanks!

**Comment by brianikim** [comment]
@andrea-ale-sbarra , could you review [7d539a7](https://github.com/liferay-commerce/liferay-portal/pull/3796/commits/7d539a7178c3efb0a1d84983475ca9a6ed2d83e0)

That commit needs to also be merged to fully address the locale issues.

Sincerely,
Brian I. Kim

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/layout/admin/util/CPDefinitionSitemapURLProvider.java [hunk 1]
```diff
@@ -156,10 +152,6 @@ protected void visitLayout(
 			return;
 		}
 
-		themeDisplay = SitemapURLProviderUtil.updateThemeDisplay(
-			_language, _portal.getLocale(themeDisplay.getRequest()),
-			themeDisplay);
-
 		String currentSiteURL = _portal.getGroupFriendlyURL(
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/layout/admin/util/CPDefinitionSitemapURLProvider.java:155]: why are you removing this method? this was added here https://issues.liferay.com/browse/COMMERCE-11452 and the updated themedisplay is used in this method call _portal.getGroupFriendlyURL, I'm not sure this is right to remove.

**Review by andrea-ale-sbarra — COMMENTED** [review]
can you please check and clarify? 

Thank you!

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/layout/admin/util/CPDefinitionSitemapURLProvider.java (hunk 1 above)
Comment (brianikim) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/layout/admin/util/CPDefinitionSitemapURLProvider.java:155]: Hi @andrea-ale-sbarra , the method is no longer necessary.

getAlternateURLs has the implementation already. 
https://github.com/liferay/liferay-portal/blob/f350a1cd720339741f646ad8041e7e0f7fec6158/portal-impl/src/com/liferay/portal/util/PortalImpl.java#L1298
https://github.com/liferay/liferay-portal/blob/f350a1cd720339741f646ad8041e7e0f7fec6158/portal-impl/src/com/liferay/portal/util/PortalImpl.java#L1403

If we update the themedisplay, we will get URLs that are the following:
"localhost:8080/en/ca/" which is not what we want. We would be essentially appending the locale twice.

Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/layout/admin/util/CPDefinitionSitemapURLProvider.java:155]: got it thanks!

**Comment by Brunno0liveira** [comment]
ci:test:commerce-functional

**Comment by pietrobocale** [comment]
New PR #3827

**Comment by pietrobocale** [comment]
ci:close

**Pull request closed without merging**